app/src/main/java/woowacourse/payments/ui/cardcatalog/CardCatalogScreen.kt

```diff
@@ -0,0 +1,73 @@
+package woowacourse.payments.ui.cardcatalog
+
+import android.app.Activity
+import android.content.Intent
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.tooling.preview.Preview
+import woowacourse.payments.ui.newcard.NewCardActivity
+import woowacourse.payments.domain.Card
+import woowacourse.payments.ui.cardcatalog.component.CardCatalogColumn
+import woowacourse.payments.ui.cardcatalog.component.CardCatalogTopBar
+import kotlin.jvm.java
+
+
+@Composable
+fun CardCatalogScreen(
+    onAddNewCardClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val context = LocalContext.current
+    val cards = remember { mutableStateListOf<Card>() }
+
+    val cardAddLauncher =
+        rememberLauncherForActivityResult(ActivityResultContracts.StartActivityForResult()) { activityResult ->
+            if (activityResult.resultCode == Activity.RESULT_OK) {
+                val newCard = activityResult.data?.getParcelableExtra<Card>("newCard")
+                newCard?.let {
+                    cards.add(newCard)
+                }
+            }
+        }
+
+    fun openAddCardWithResult() {
+        val intent = Intent(context, NewCardActivity::class.java)
+        cardAddLauncher.launch(intent)
+    }
+
+    Scaffold(
+        modifier = modifier,
+        topBar = {
+            CardCatalogTopBar(
+                cards = cards  ,
+                onAddNewCardClick = {
+                    onAddNewCardClick()
+                    openAddCardWithResult()
+                })
```

Comment:
```suggestion
                onAddNewCardClick = {
                    openAddCardWithResult()
                })
```

onAddNewCardClick()는 없어도 NewCardActivity로 잘 이동하는 것 같네요. 바로 아래(60번 라인)도 마찬가지입니당 원인이 무엇일까용...?

Reply:
[refactor: activity에서 이동 로직 삭제](https://github.com/woowacourse/android-payments/pull/46/commits/3aff7b5f78c250acbb891be81cda4577577e2fb1)

두개다 새로운 카드를 만들기 위해서 이동하는 로직이네용! 그래서 `onAddNewCardClick()`가 없어도 잘 작동이 되네용...